pkgs/applications/science/math/caffe/default.nix

```diff
@@ -1,27 +1,50 @@
-{ stdenv, lib
+{ stdenv, lib, runCommand
 , fetchFromGitHub
+, fetchurl
 , cmake
 , boost
 , google-gflags
 , glog
 , hdf5-cpp
-, leveldb
-, lmdb
 , opencv3
 , protobuf
-, snappy
 , doxygen
 , openblas
-, cudaSupport ? true, cudatoolkit
+, Accelerate, CoreGraphics, CoreVideo
+, lmdbSupport ? true, lmdb
+, leveldbSupport ? true, leveldb, snappy
+, cudaSupport ? stdenv.isLinux, cudatoolkit
 , cudnnSupport ? false, cudnn ? null
 , ncclSupport ? false, nccl ? null
 , pythonSupport ? false, python ? null, numpy ? null
 }:
 
+assert leveldbSupport -> (leveldb != null && snappy != null);
 assert cudnnSupport -> cudaSupport;
 assert ncclSupport -> cudaSupport;
 assert pythonSupport -> (python != null && numpy != null);
 
+let
+  toggle = bool: if bool then "ON" else "OFF";
+
+  test_model_weights = fetchurl {
+    url = "http://dl.caffe.berkeleyvision.org/bvlc_reference_caffenet.caffemodel";
+    sha256 = "472d4a06035497b180636d8a82667129960371375bd10fcb6df5c6c7631f25e0";
+  };
+
+  # work around /usr/lib having different lib names than the framework
+  macblas = runCommand "macblas" {
+    propagatedBuildInputs = [ Accelerate ];
+  } ''
+    mkdir -p $out/lib
+    ln -s /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib $out/lib/libblas.dylib
```

Comment:
This should link to the Accelerate derivation, not /System.

Reply:
How do you solve an empty `${Accelerate}/Library/Frameworks/Accelerate.framework/Versions/A` ?

Reply:
I think the logic is already handled here:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/darwin/apple-sdk/default.nix#L77-L131

Although maybe it fails here...

Reply:
It was using openblas but cmake decided to use Accelerate even when it's not provided. Maybe the best option is to patch the build system.